src/main/java/seedu/task/logic/commands/ClearCommand.java

```diff
@@ -1,22 +1,69 @@
  */
 public class ClearCommand extends Command {
 
     public static final String COMMAND_WORD = "clear";
-    public static final String MESSAGE_SUCCESS = "Address book has been cleared!";
+    public static final String MESSAGE_SUCCESS = "All %s %s has been cleared!";
+    public static final String MESSAGE_COMPLETED = "completed";
+    public static final String MESSAGE_COMPLETED_UNCOMPLETED = "completed and uncompleted";
+    public static final String MESSAGE_TASKS = "tasks";
+    public static final String MESSAGE_EVENTS = "events";
+    public static final String MESSAGE_TASKS_EVENTS = "tasks and events";
+    public static final String MESSAGE_USAGE = "adsa ";
+    
+    private final String isTask;
+    private final boolean isAll;
 
-    public ClearCommand() {}
+    public ClearCommand(String tag_1, boolean tag_2) {
+        this.isTask = tag_1;
+        this.isAll = tag_2;
+    }
 
 
     @Override
     public CommandResult execute() {
-        assert model != null;
-        model.resetData(TaskBook.getEmptyTaskBook());
-        return new CommandResult(MESSAGE_SUCCESS);
+        
+        if(isTask == "t & e" && !isAll){ // clears completed tasks and events
+            model.clearTasks();
+            model.clearEvents();
+            return new CommandResult(String.format(MESSAGE_SUCCESS, MESSAGE_COMPLETED, MESSAGE_TASKS_EVENTS));
+        }else if (isTask == "t" && !isAll){ // clears completed tasks
+            model.clearTasks();
+            return new CommandResult(String.format(MESSAGE_SUCCESS, MESSAGE_COMPLETED, MESSAGE_TASKS));
+        }else if (isTask == "e" && !isAll){ // clears completed events
+            model.clearEvents();
+            return new CommandResult(String.format(MESSAGE_SUCCESS, MESSAGE_COMPLETED, MESSAGE_EVENTS));
+        }else if (isTask == "t" && isAll){ // clears all completed and uncompleted tasks
+            assert model != null;
+            ReadOnlyTaskBook taskbook = model.getTaskBook();
+            model.resetData(new TaskBook(new UniqueTaskList(), taskbook.getUniqueEventList()));
+            return new CommandResult(String.format(MESSAGE_SUCCESS, MESSAGE_COMPLETED_UNCOMPLETED, MESSAGE_TASKS));
+        }else if (isTask == "e" && isAll){ // clears all completed and uncompleted events
+            assert model != null;
+            ReadOnlyTaskBook taskbook = model.getTaskBook();
+            model.resetData(new TaskBook(taskbook.getUniqueTaskList(), new UniqueEventList()));
+            return new CommandResult(String.format(MESSAGE_SUCCESS, MESSAGE_COMPLETED_UNCOMPLETED, MESSAGE_EVENTS));
+        }else if (isTask == "t & e" && isAll){ // clears all completed and uncompleted tasks and events
+            assert model != null;
+            model.resetData(TaskBook.getEmptyTaskBook());
+            return new CommandResult(String.format(MESSAGE_SUCCESS, MESSAGE_COMPLETED_UNCOMPLETED, MESSAGE_TASKS_EVENTS));
+        }
+        return null;       
```

Comment:
I wanted to put the invalid command, however I realised that even though I can return a commandresult that has a string saying it is an invalid command, it wouldn't post an invalid command event. The invalid command event is initiated in the parser and not in the command's execution.

I was thinking of doing an assertion that my type is either of the 3, another way would be to create an enumerator which means it can only take this possible 3 values. This would prevent eclipse from flagging this as an error. Do let me know what you think, we can discuss more tmr!